libs/payments/ui/src/lib/client/components/SubscriptionContent/index.tsx

```diff
@@ -64,8 +93,144 @@ export const SubscriptionContent = ({
     <>
+      <Dialog.Root open={openResubscribeSuccessDialog}>
+        <Dialog.Portal>
+          <Dialog.Overlay className='fixed inset-0 bg-black/75 z-50' />
+          <Dialog.Content
+            className='w-11/12 max-w-[545px] text-center px-7 pt-6 pb-8 rounded-xl shadow inline-block fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 z-9999 bg-white'
+            onEscapeKeyDown={() => setOpenResubscribeSuccessDialog(false)}
+            onPointerDownOutside={() => setOpenResubscribeSuccessDialog(false)}
+            onInteractOutside={() => setOpenResubscribeSuccessDialog(false)}
+          >
+            <Dialog.Title className='font-bold leading-6 m-5 space-y-5'>
+              <Image src={webIcon} alt={productName} height={64} width={64} className='h-16 w-16 mx-auto' />
+              <Localized
+                id="resubscribe-success-dialog-title"
+              >
+                <p>Thanks! You’re all set.</p>
+              </Localized>
+            </Dialog.Title>
+            <Dialog.Description className='leading-6 space-y-4'>
+              <Localized
+                id="resubscribe-success-dialog-action-button"
+              >
+                <SubmitButton
+                  className="h-10 w-full"
+                  variant={ButtonVariant.Primary}
+                  onClick={() => setOpenResubscribeSuccessDialog(false)}
+                  aria-label={`Stay subscribed to ${productName}`}
+                >
+                  Close
+                </SubmitButton>
+              </Localized>
+            </Dialog.Description>
+            <Dialog.Close asChild
+            >
+              <button
+                className="absolute bg-transparent border-0 cursor-pointer flex items-center justify-center w-6 h-6 m-0 p-0 top-4 right-4 hover:bg-grey-200 hover:rounded focus:border-blue-400 focus:rounded focus:shadow-input-blue-focus after:absolute after:content-[''] after:top-0 after:left-0 after:w-full after:h-full after:bg-white after:opacity-50 after:z-10"
+                onClick={() => setOpenResubscribeSuccessDialog(false)}
+              >
+                <Image src={CloseIcon} alt={l10n.getString('dialog-close', null, 'Close dialog')} />
+              </button>
+            </Dialog.Close>
+          </Dialog.Content>
+        </Dialog.Portal>
+      </Dialog.Root>
+
+      <Dialog.Root open={openResubscribeDialog}>
+        <Dialog.Portal>
+          <Dialog.Overlay className='fixed inset-0 bg-black/75 z-50' />
+          <Dialog.Content
+            className='w-11/12 max-w-[545px] text-center px-7 pt-6 pb-8 rounded-xl shadow inline-block fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 z-9999 bg-white'
+            onEscapeKeyDown={() => setOpenResubscribeDialog(false)}
+            onPointerDownOutside={() => setOpenResubscribeDialog(false)}
+            onInteractOutside={() => setOpenResubscribeDialog(false)}
+          >
+            <Dialog.Title className='font-bold leading-6 m-5 space-y-5'>
```

Comment:
Suggestion: improvements for accessibility
`id="resubscribe-dialog-title"` to `Dialog.Title`